packages/framework/src/api/data.ts

```diff
@@ -11,6 +12,8 @@ import { screenDataAtom, type ScreenContextDefinition } from "../state";
 import { isUsingMockResponse } from "../appConfig";
 import { mockResponse } from "../evaluate/mock";
 
+const queryClient = new QueryClient();
```

Comment:
You need to use the same instance of the queryClient otherwise it will not be the same cache.